src/lib.rs

```diff
@@ -30,12 +33,14 @@ pub mod package;
 /// Supported registries
 pub mod registry;
 
+mod managed_file;
```

Comment:
Can rename this to smth more generic? it doesnt seem sensible to have a whole mod for _one_ enum. Maybe this can be just toplevel in the lib and be pub(crate)?

Reply:
I think it'd be best to make a module of this and move the `manifest.rs`, `credentials.rs` and `lock.rs` modules there. Happy to move this to the lib mod for now but does that seem sensible?

Reply:
Yes, i agree! Once we merge them, lets put it there